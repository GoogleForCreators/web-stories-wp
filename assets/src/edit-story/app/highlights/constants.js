/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * External dependencies
 */
import { css } from 'styled-components';

const FLASH_KEYFRAMES = `
@keyframes flash {
  50% {
    opacity: 50%;
  }
}
`;

const FLASH_ANIMATION = `
  animation: flash 0.25s ease-out 3;
`;

export const panelRowStyles = css`
  position: relative;
  &:after {
    display: block;
    position: absolute;
    top: -10px;
    right: -10px;
    bottom: -10px;
    left: -10px;
    content: '';
    background-color: ${({ theme }) => theme.colors.callout};
    opacity: 0;
    pointer-events: none;
    ${FLASH_ANIMATION}
  }
  ${FLASH_KEYFRAMES}
`;

const defaultBehavior = { css: panelRowStyles };

export const INPUTS = {
  storyTitle: { ...defaultBehavior, focus: true },
  pageMenu: defaultBehavior,
  document: {
    publisherLogo: defaultBehavior,
    cover: defaultBehavior,
    excerpt: { ...defaultBehavior, focus: true },
  },
  design: {
    captions: defaultBehavior,
    assistiveText: {
      ...defaultBehavior,
      focus: true,
    },
  },
  tab: {
    text: defaultBehavior,
  },
};
