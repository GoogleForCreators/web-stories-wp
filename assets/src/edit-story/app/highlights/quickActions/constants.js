/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { __ } from '@web-stories-wp/i18n';

export const ELEMENT_TYPE = {
  IMAGE: 'image',
  SHAPE: 'shape',
  TEXT: 'text',
  VIDEO: 'video',
  GIF: 'gif',
  BACKGROUND: 'background',
};

export const ACTION_TEXT = {
  ADD_ANIMATION: __('Add animation', 'web-stories'),
  ADD_LINK: __('Add link', 'web-stories'),
  ADD_CAPTIONS: __('Add captions', 'web-stories'),
  CHANGE_BACKGROUND_COLOR: __('Change background color', 'web-stories'),
  CHANGE_TEXT_COLOR: __('Change color', 'web-stories'),
  CHANGE_COLOR: __('Change color', 'web-stories'),
  RESET_ELEMENT: __('Reset element', 'web-stories'),
  CHANGE_FONT: __('Edit text', 'web-stories'),
  INSERT_BACKGROUND_MEDIA: __('Insert background media', 'web-stories'),
  INSERT_TEXT: __('Insert text', 'web-stories'),
  REPLACE_BACKGROUND_MEDIA: __('Replace background', 'web-stories'),
  REPLACE_MEDIA: __('Replace media', 'web-stories'),
};

export const RESET_PROPERTIES = {
  ANIMATION: 'animation',
  OVERLAY: 'overlay',
  STYLES: 'styles',
};

export const RESET_DEFAULTS = {
  TEXT_BORDER_RADIUS: {
    locked: true,
    topLeft: 2,
    topRight: 2,
    bottomLeft: 2,
    bottomRight: 2,
  },
  STANDARD_BORDER_RADIUS: 0,
};
