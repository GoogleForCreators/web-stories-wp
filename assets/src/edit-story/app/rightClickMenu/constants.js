/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * External dependencies
 */
import { _x, __ } from '@web-stories-wp/i18n';

export const RIGHT_CLICK_MENU_LABELS = {
  BRING_FORWARD: __('Bring forward', 'web-stories'),
  BRING_TO_FRONT: __('Bring to front', 'web-stories'),
  COPY: __('Copy', 'web-stories'),
  DELETE_PAGE: __('Delete page', 'web-stories'),
  DELETE: __('Delete', 'web-stories'),
  DUPLICATE_PAGE: __('Duplicate page', 'web-stories'),
  PASTE: __('Paste', 'web-stories'),
  SCALE_AND_CROP_IMAGE: __('Scale & crop image', 'web-stories'),
  SEND_BACKWARD: __('Send backward', 'web-stories'),
  SEND_TO_BACK: __('Send to back', 'web-stories'),
  SET_AS_PAGE_BACKGROUND: __('Set as page background', 'web-stories'),
};

export const RIGHT_CLICK_MENU_SHORTCUT_LABELS = {
  COMMAND_C: _x(
    'Command C',
    'The keyboard keys "Command" and "C"',
    'web-stories'
  ),
  CONTROL_C: _x(
    'Control C',
    'The keyboard keys "Control" and "C"',
    'web-stories'
  ),
  COMMAND_V: _x(
    'Command V',
    'The keyboard keys "Command" and "V"',
    'web-stories'
  ),
  CONTROL_V: _x(
    'Control V',
    'The keyboard keys "Control" and "V"',
    'web-stories'
  ),
  DELETE: _x('Delete', 'The keyboard key "Delete"', 'web-stories'),
  COMMAND_OPEN_BRACKET: _x(
    'Command Open Bracket',
    'The keyboard keys "Command" and "Open Bracket"',
    'web-stories'
  ),
  CONTROL_OPEN_BRACKET: _x(
    'Control Open Bracket',
    'The keyboard keys "Control" and "Open Bracket"',
    'web-stories'
  ),
  OPTION_COMMAND_OPEN_BRACKET: _x(
    'Option Command Open Bracket',
    'The keyboard keys "Option", "Command", and "Open Bracket"',
    'web-stories'
  ),
  OPTION_CONTROL_OPEN_BRACKET: _x(
    'Option Control Open Bracket',
    'The keyboard keys "Option", "Control", and "Open Bracket"',
    'web-stories'
  ),
  COMMAND_CLOSE_BRACKET: _x(
    'Command Close Bracket',
    'The keyboard keys "Command" and "Close Bracket"',
    'web-stories'
  ),
  CONTROL_CLOSE_BRACKET: _x(
    'Control Close Bracket',
    'The keyboard keys "Control" and "Close Bracket"',
    'web-stories'
  ),
  OPTION_COMMAND_CLOSE_BRACKET: _x(
    'Option Command Close Bracket',
    'The keyboard keys "Option", "Command", and "Close Bracket"',
    'web-stories'
  ),
  OPTION_CONTROL_CLOSE_BRACKET: _x(
    'Option Control Close Bracket',
    'The keyboard keys "Option", "Control", and "Close Bracket"',
    'web-stories'
  ),
};

export const RIGHT_CLICK_SNACKBAR_TEXT = {
  ADDED_STYLE: __('Added style to "Saved styles".', 'web-stories'),
  ADDED_COLOR: __('Added color to "Saved colors".', 'web-stories'),
  COPIED_STYLE: __('Copied style.', 'web-stories'),
  PASTED_STYLE: __('Pasted style.', 'web-stories'),
  PROPERTIES_COPIED: __('Properties copied.', 'web-stories'),
  PROPERTIES_PASTED: __('Properties pasted.', 'web-stories'),
  TEXT_COPIED: __('Text copied.', 'web-stories'),
  TEXT_PASTED: __('Copied styles.', 'web-stories'),
  UNDO: __('Undo', 'web-stories'),
};
