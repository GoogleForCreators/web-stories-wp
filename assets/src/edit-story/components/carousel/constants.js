/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { THEME_CONSTANTS } from '@web-stories-wp/design-system';

// If the workspace is wider than this limit, this margin will be applied to the right
export const VERY_WIDE_WORKSPACE_LIMIT = 1000;
export const VERY_WIDE_MARGIN = 8;

// The buttons and menu gutters
export const BUTTON_WIDTH = THEME_CONSTANTS.ICON_SIZE;
export const BUTTON_GAP = 8;
export const BUTTON_GUTTER = BUTTON_WIDTH + BUTTON_GAP;
export const MENU_GUTTER = 167;

// Thumbnail size varies with workspace size - over or under this limit
export const WIDE_WORKSPACE_LIMIT = 700;
export const WIDE_THUMBNAIL_WIDTH = 40;
export const WIDE_THUMBNAIL_HEIGHT = 60;
export const NARROW_THUMBNAIL_WIDTH = 36;
export const NARROW_THUMBNAIL_HEIGHT = 54;
export const THUMBNAIL_MARGIN = 16;
export const THUMBNAIL_LINE_WIDTH = 4;
