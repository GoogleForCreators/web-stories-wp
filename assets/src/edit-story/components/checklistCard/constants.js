/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export const CARD_TYPE = {
  SINGLE_ISSUE: 'single_issue',
  MULTIPLE_ISSUE: 'multiple_issue',
};
export const GRID_VARIANT = {
  SINGLE_WITH_THUMBNAIL: 'single_with_thumbnail',
  SINGLE: 'single',
  DEFAULT: 'single',
};
