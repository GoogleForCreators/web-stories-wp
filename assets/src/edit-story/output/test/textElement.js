/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
jest.mock('flagged');
import { useFeature } from 'flagged';
import { renderToStaticMarkup } from 'react-dom/server';
import PropTypes from 'prop-types';

/**
 * Internal dependencies
 */
import { StoryAnimation } from '../../../animation';
import { BACKGROUND_TEXT_MODE } from '../../constants';
import OutputElement from '../element';
import { DEFAULT_TEXT } from './_utils/constants';

function WrapAnimation({ children }) {
  return (
    <StoryAnimation.Provider animations={[]}>
      {children}
    </StoryAnimation.Provider>
  );
}

WrapAnimation.propTypes = {
  children: PropTypes.node,
};

const TEXT_WITH_COLOR = {
  ...DEFAULT_TEXT,
  content:
    '<span style="color: #0a08ec">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</span>',
};

const TEXT_WITH_BIU = {
  ...DEFAULT_TEXT,
  content:
    '<span style="font-weight: 700; font-style: italic; text-decoration: underline">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</span>',
};

const TEXT_WITH_FILL = {
  ...DEFAULT_TEXT,
  backgroundTextMode: BACKGROUND_TEXT_MODE.FILL,
};

const TEXT_WITH_HIGHLIGHT = {
  ...DEFAULT_TEXT,
  backgroundTextMode: BACKGROUND_TEXT_MODE.HIGHLIGHT,
};

const TEXT_WITH_PADDING = {
  ...DEFAULT_TEXT,
  padding: {
    locked: false,
    horizontal: 12,
    vertical: 25,
  },
};

const TEXT_WITH_FONTSIZE_FAMILY = {
  ...DEFAULT_TEXT,
  font: {
    ...DEFAULT_TEXT.font,
    family: 'Bungee Shade',
  },
  fontSize: 45,
};

const TEXT_WITH_LINEHEIGHT = {
  ...DEFAULT_TEXT,
  lineHeight: 4,
};

const TEXT_WITH_LETTERSPACING = {
  ...DEFAULT_TEXT,
  content:
    '<span style="letter-spacing: 0.5em">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</span>',
};

const TEXT_WITH_ROTATION = {
  ...DEFAULT_TEXT,
  rotationAngle: 45,
};

describe('Text Element output', () => {
  useFeature.mockImplementation((feature) => {
    const config = {
      enableAnimation: false,
    };

    return config[feature];
  });

  it('should render text with color', () => {
    const element = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_WITH_COLOR} />
      </WrapAnimation>
    );
    expect(element).toMatchSnapshot();
  });

  it('should render text with fill', () => {
    const element = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_WITH_FILL} />
      </WrapAnimation>
    );
    expect(element).toMatchSnapshot();
  });

  it('should render text with highlight', () => {
    const element = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_WITH_HIGHLIGHT} />
      </WrapAnimation>
    );
    expect(element).toMatchSnapshot();
  });

  it('should render text with padding', () => {
    const element = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_WITH_PADDING} />
      </WrapAnimation>
    );
    expect(element).toMatchSnapshot();
  });

  it('should render text with alignment', () => {
    const TEXT_ALIGN_LEFT = {
      ...DEFAULT_TEXT,
      textAlign: 'left',
    };

    const textLeft = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_ALIGN_LEFT} />
      </WrapAnimation>
    );
    expect(textLeft).toMatchSnapshot();

    const TEXT_ALIGN_RIGHT = {
      ...DEFAULT_TEXT,
      textAlign: 'right',
    };

    const textRight = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_ALIGN_RIGHT} />
      </WrapAnimation>
    );
    expect(textRight).toMatchSnapshot();

    const TEXT_ALIGN_CENTER = {
      ...DEFAULT_TEXT,
      textAlign: 'center',
    };

    const textCenter = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_ALIGN_CENTER} />
      </WrapAnimation>
    );
    expect(textCenter).toMatchSnapshot();

    const TEXT_ALIGN_JUSTIFY = {
      ...DEFAULT_TEXT,
      textAlign: 'justify',
    };

    const textJustify = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_ALIGN_JUSTIFY} />
      </WrapAnimation>
    );
    expect(textJustify).toMatchSnapshot();
  });

  it('should render text with bold, italic, and underline', () => {
    const element = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_WITH_BIU} />
      </WrapAnimation>
    );
    expect(element).toMatchSnapshot();
  });

  it('should render text with adjusted font-size and family', () => {
    const element = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_WITH_FONTSIZE_FAMILY} />
      </WrapAnimation>
    );
    expect(element).toMatchSnapshot();
  });

  it('should render text with line-height', () => {
    const element = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_WITH_LINEHEIGHT} />
      </WrapAnimation>
    );
    expect(element).toMatchSnapshot();
  });

  it('should render text with letter-spacing', () => {
    const element = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_WITH_LETTERSPACING} />
      </WrapAnimation>
    );
    expect(element).toMatchSnapshot();
  });

  it('should render text with applied rotation', () => {
    const element = renderToStaticMarkup(
      <WrapAnimation>
        <OutputElement element={TEXT_WITH_ROTATION} />
      </WrapAnimation>
    );
    expect(element).toMatchSnapshot();
  });

  describe('AMP validation', () => {
    it('should produce valid AMP output when setting text color', async () => {
      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_WITH_COLOR} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();
    });

    it('should produce valid AMP output when setting text fill', async () => {
      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_WITH_FILL} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();
    });

    it('should produce valid AMP output when setting text highlight', async () => {
      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_WITH_HIGHLIGHT} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();
    });

    it('should produce valid AMP output when setting text padding', async () => {
      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_WITH_PADDING} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();
    });

    it('should produce valid AMP output when setting text alignment', async () => {
      const TEXT_ALIGN_LEFT = {
        ...DEFAULT_TEXT,
        textAlign: 'left',
      };

      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_ALIGN_LEFT} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();

      const TEXT_ALIGN_RIGHT = {
        ...DEFAULT_TEXT,
        textAlign: 'right',
      };

      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_ALIGN_RIGHT} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();

      const TEXT_ALIGN_CENTER = {
        ...DEFAULT_TEXT,
        textAlign: 'center',
      };

      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_ALIGN_CENTER} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();

      const TEXT_ALIGN_JUSTIFY = {
        ...DEFAULT_TEXT,
        textAlign: 'justify',
      };

      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_ALIGN_JUSTIFY} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();
    });

    it('should produce valid AMP output when setting text to bold, italic, and underline', async () => {
      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_WITH_BIU} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();
    });

    it('should produce valid AMP output when setting text font-size and family', async () => {
      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_WITH_FONTSIZE_FAMILY} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();
    });

    it('should produce valid AMP output when setting text line-height', async () => {
      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_WITH_LINEHEIGHT} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();
    });

    it('should produce valid AMP output when setting text letter-spacing', async () => {
      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_WITH_LETTERSPACING} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();
    });

    it('should produce valid AMP output when setting text rotation', async () => {
      await expect(
        <WrapAnimation>
          <OutputElement element={TEXT_WITH_ROTATION} />
        </WrapAnimation>
      ).toBeValidAMPStoryElement();
    });
  });
});
