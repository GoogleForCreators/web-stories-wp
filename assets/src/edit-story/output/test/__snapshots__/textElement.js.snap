// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Element output should render text with adjusted font-size and family 1`] = `"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:-3.5838895631067964% 0;font-family:&quot;Bungee Shade&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.728155em;line-height:1.5;text-align:initial;padding:0% 0%;color:#000000\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p></div></div>"`;

exports[`Text Element output should render text with alignment 1`] = `"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:-1.4335558252427185% 0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:1.5;text-align:left;padding:0% 0%;color:#000000\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p></div></div>"`;

exports[`Text Element output should render text with alignment 2`] = `"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:-1.4335558252427185% 0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:1.5;text-align:right;padding:0% 0%;color:#000000\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p></div></div>"`;

exports[`Text Element output should render text with alignment 3`] = `"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:-1.4335558252427185% 0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:1.5;text-align:center;padding:0% 0%;color:#000000\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p></div></div>"`;

exports[`Text Element output should render text with alignment 4`] = `"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:-1.4335558252427185% 0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:1.5;text-align:justify;padding:0% 0%;color:#000000\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p></div></div>"`;

exports[`Text Element output should render text with applied rotation 1`] = `"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;transform:rotate(45deg);opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:-1.4335558252427185% 0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:1.5;text-align:initial;padding:0% 0%;color:#000000\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p></div></div>"`;

exports[`Text Element output should render text with bold, italic, and underline 1`] = `"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:-1.4335558252427185% 0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:1.5;text-align:initial;padding:0% 0%;color:#000000\\"><span style=\\"font-weight: 700; font-style: italic; text-decoration: underline\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</span></p></div></div>"`;

exports[`Text Element output should render text with color 1`] = `"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:-1.4335558252427185% 0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:1.5;text-align:initial;padding:0% 0%;color:#000000\\"><span style=\\"color: #0a08ec\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</span></p></div></div>"`;

exports[`Text Element output should render text with fill 1`] = `"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:-1.4335558252427185% 0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:1.5;text-align:initial;padding:0% 0%;background-color:#c4c4c4;color:#000000\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p></div></div>"`;

exports[`Text Element output should render text with highlight 1`] = `
"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:calc(
    1.5em
    -
    0em
  );text-align:initial;padding:0;background-color:#c4c4c4;color:#000000;background:none\\"><span style=\\"display:block;position:relative;left:-0%;top:0;margin:-1.4335558252427185% 0\\"><span style=\\"background-color:#c4c4c4;-webkit-box-decoration-break:clone;box-decoration-break:clone;position:relative;padding:0% 0%;color:transparent\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</span></span></p><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:calc(
    1.5em
    -
    0em
  );text-align:initial;padding:0;background-color:#c4c4c4;color:#000000;background:none;position:absolute;top:0\\" aria-hidden=\\"true\\"><span style=\\"display:block;position:relative;left:-0%;top:0;margin:-1.4335558252427185% 0\\"><span style=\\"background-color:#c4c4c4;-webkit-box-decoration-break:clone;box-decoration-break:clone;position:relative;padding:0% 0%;background:none\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</span></span></p></div></div>"
`;

exports[`Text Element output should render text with letter-spacing 1`] = `"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:-1.4335558252427185% 0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:1.5;text-align:initial;padding:0% 0%;color:#000000\\"><span style=\\"letter-spacing: 0.5em\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</span></p></div></div>"`;

exports[`Text Element output should render text with line-height 1`] = `"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:-12.355885922330097% 0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:4;text-align:initial;padding:0% 0%;color:#000000\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p></div></div>"`;

exports[`Text Element output should render text with padding 1`] = `"<div style=\\"position:absolute;pointer-events:none;left:9.70874%;top:48.54369%;width:50%;height:12.13592%;opacity:1\\"><div style=\\"pointer-events:initial;width:100%;height:100%;display:block;position:absolute;top:0;left:0\\" class=\\"mask\\" id=\\"el-8d110168-e765-4c76-b63a-74ec23d54f27\\"><p class=\\"fill\\" style=\\"white-space:pre-wrap;overflow-wrap:break-word;word-break:break-word;margin:-1.4335558252427185% 0;font-family:&quot;Roboto&quot;,&quot;Helvetica Neue&quot;,&quot;Helvetica&quot;,sans-serif;font-size:0.291262em;line-height:1.5;text-align:initial;padding:12.135922330097088% 5.825242718446602%;color:#000000\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p></div></div>"`;
